cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -2234,16 +2234,22 @@ if((NOT ARROW_SIMD_LEVEL STREQUAL "NONE") OR (NOT ARROW_RUNTIME_SIMD_LEVEL STREQ
 else()
   set(ARROW_USE_XSIMD FALSE)
 endif()
+
+set(ARROW_XSIMD_REQUIRED_VERSION "8.1.0")
+
 if(ARROW_USE_XSIMD)
-  set(xsimd_SOURCE "BUNDLED")
-  resolve_dependency(xsimd)
+  resolve_dependency(xsimd REQUIRED_VERSION ${ARROW_XSIMD_REQUIRED_VERSION})
 
-  add_library(xsimd INTERFACE IMPORTED)
-  if(CMAKE_VERSION VERSION_LESS 3.11)
-    set_target_properties(xsimd PROPERTIES INTERFACE_INCLUDE_DIRECTORIES
-                                           "${XSIMD_INCLUDE_DIR}")
+  if(xsimd_SOURCE STREQUAL "BUNDLED")
+    add_library(xsimd INTERFACE IMPORTED)
+    if(CMAKE_VERSION VERSION_LESS 3.11)
+      set_target_properties(xsimd PROPERTIES INTERFACE_INCLUDE_DIRECTORIES
+                                             "${XSIMD_INCLUDE_DIR}")
+    else()
+      target_include_directories(xsimd INTERFACE "${XSIMD_INCLUDE_DIR}")
+    endif()
```

Comment:
Curious why need to check version and call two different functions.
`target_include_directories` is not supported before cmake 3.11?
Can we use `set_target_properties` on all version?

Some code doesn't check version and only calls `target_include_directores`. Is it a problem?
e.g. https://github.com/apache/arrow/blob/master/cpp/cmake_modules/ThirdpartyToolchain.cmake#L4631

Reply:
`target_include_directories()` in 3.10 or earlier doesn't support `INTERFACE` against `IMPORTED` target:

https://cmake.org/cmake/help/latest/command/target_include_directories.html

> New in version 3.11: Allow setting INTERFACE items on [IMPORTED targets](https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#imported-targets).

For `ucx::ucx` case, we need to add similar version check. It may be better that we add `arrow_imported_target_interface_include_directories()` or something and use it.

Reply:
Thanks for the explanation.
I've created a jira https://issues.apache.org/jira/browse/ARROW-16751